cfme/cloud/host_aggregates.py

```diff
@@ -40,29 +87,102 @@ class my_filters(Accordion):  # noqa
         navigation = BootstrapNav('.//div/ul')
         tree = ManageIQTree()
 
+    @property
+    def is_displayed(self):
+        return (
+            self.in_host_aggregates and
+            self.entities.title.text == 'Host Aggregates')
 
-class HostAggregatesDefaultView(HostAggregatesView):
-    title = Text('//div[@id="main-content"]//h1')
 
+class HostAggregatesDetailsView(HostAggregatesView):
     @property
     def is_displayed(self):
-        return self.in_host_aggregates and self.title.is_displayed
+        obj = self.context['object']
+        return (
+            self.in_host_aggregates and
+            self.entities.title.text == obj.expected_details_title and
+            self.entities.breadcrumb.active_location == obj.expected_details_breadcrumb and
+            self.entities.relationships.get_text_of('Cloud Provider') == obj.provider.name
+        )
+
+    toolbar = HostAggregatesDetailsToolBar()
+    sidebar = HostAggregatesDetailsAccordion()
+    entities = HostAggregatesDetailsEntities()
 
 
 @attr.s
-class HostAggregates(BaseEntity):
-    pass
+class HostAggregates(BaseEntity, Taggable):
+    """
+    Host Aggregates class to support navigation
+    """
+    _param_name = "HostAggregate"
+
+    name = attr.ib()
+    provider = attr.ib()
+    ram = attr.ib(default=None)
+    vcpus = attr.ib(default=None)
+    disk = attr.ib(default=None)
+    swap = attr.ib(default=None)
+    rxtx = attr.ib(default=None)
+    is_public = attr.ib(default=True)
+    tenant = attr.ib(default=None)
+
+    def refresh(self):
+        """Refresh provider relationships and browser"""
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def instance_count(self):
+        """ number of instances using host aggregates.
+
+        Returns:
+            :py:class:`int` instance count.
+        """
+        view = navigate_to(self, 'Details')
+        return int(view.entities.relationships.get_text_of('Instances'))
 
 
 @attr.s
-class HostAggregatesCollection(BaseCollection):
+class HostAggregatesCollection(BaseCollection, TaggableCollection):
     ENTITY = HostAggregates
 
+    def all(self):
+        provider = self.filters.get('provider')  # None if no filter, need for entity instantiation
+        view = navigate_to(self, 'All')
+        result = []
+        for _ in view.entities.paginator.pages():
```

Comment:
Rather than looping over all the pages, you can use,
`flavors = view.entities.get_all(surf_pages=True)`

That will loop through all the pages for you.